apps/studio/src/server/modules/site/site.router.ts

```diff
@@ -90,6 +92,65 @@ export const siteRouter = router({
       })
       return getSiteConfig(db, id)
     }),
+  updateSiteConfig: protectedProcedure
+    .input(updateSiteConfigSchema)
+    .mutation(async ({ ctx, input: { siteId, siteName } }) => {
+      await validateUserPermissionsForSite({
+        siteId,
+        userId: ctx.user.id,
+        action: "update",
+      })
+      const {
+        user: { id: userId },
+      } = ctx
+
+      const user = await db
+        .selectFrom("User")
+        .where("id", "=", userId)
+        .selectAll()
+        .executeTakeFirstOrThrow()
+
+      // TODO: audit log
+      const updatedConfig = await db.transaction().execute(async (tx) => {
+        const site = await tx
+          .selectFrom("Site")
+          .where("id", "=", siteId)
+          .selectAll()
+          .executeTakeFirstOrThrow()
+
+        const { config } = site
+
+        const updatedSite = await tx
+          .updateTable("Site")
+          .set({ config: jsonb({ ...config, siteName }) })
+          .where("id", "=", siteId)
+          .returningAll()
+          .executeTakeFirstOrThrow()
+
+        const { config: updatedConfig } = updatedSite
+
+        if (updatedConfig.search?.type === "searchSG")
+          void updateSearchSGConfig(
+            { name: siteName },
+            updatedConfig.search.clientId,
+            new URL(updatedConfig.url),
+          )
```

Comment:
issue: not a good practice to do 3rd party API call in a DB transaction

Reply:
sure, i'll shift out!